taskvine/src/manager/vine_manager.c

```diff
@@ -4769,9 +4781,35 @@ void vine_get_stats(struct vine_manager *q, struct vine_stats *s)
 	// s->workers_able computed below.
 
 	//info about tasks
-	s->tasks_waiting      = task_state_count(q, NULL, VINE_TASK_READY);
-	s->tasks_with_results = task_state_count(q, NULL, VINE_TASK_WAITING_RETRIEVAL);
-	s->tasks_on_workers   = task_state_count(q, NULL, VINE_TASK_RUNNING) + s->tasks_with_results;
+	struct vine_task *t;
```

Comment:
Now that we have separate lists for ready, waiting, retrieved, can you just use `list_size` to return the count of tasks in each state, rather than iterating over everything?